faststream/redis/subscriber/usecases/stream_subscriber.py

```diff
@@ -96,38 +96,145 @@ async def start(self) -> None:
 
         stream = self.stream_sub
 
-        read: ReadCallable
+        read: Callable[
+            [str],
+            Awaitable[
+                tuple[
+                    tuple[
+                        TopicName,
+                        tuple[
+                            tuple[
+                                Offset,
+                                dict[bytes, bytes],
+                            ],
+                            ...,
+                        ],
+                    ],
+                    ...,
+                ],
+            ],
+        ]
 
         if stream.group and stream.consumer:
-            group_create_id = "$" if self.last_id == ">" else self.last_id
-            try:
-                await client.xgroup_create(
-                    name=stream.name,
-                    id=group_create_id,
-                    groupname=stream.group,
-                    mkstream=True,
+            await self._create_group()
+
+            def read(
+                _: str,
+            ) -> Awaitable[
+                tuple[
+                    tuple[
+                        TopicName,
+                        tuple[
+                            tuple[
+                                Offset,
+                                dict[bytes, bytes],
+                            ],
+                            ...,
+                        ],
+                    ],
+                    ...,
+                ],
+            ]:
+                async def read_from_group() -> tuple[
```

Comment:
i'm thinking at this point you may as well just make `def read` an async function and do without the nested `read_from_group` it may make things more readable ?

Reply:
this type annotation is likely the first conflict, you can use `ReadResponse` as a replacement